### DISCUSSION TITLE
question on rtxmlrpc usage

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#49] [Q&A] [answered]
Hello,

I am reading up on the documentation to set things up in my new box.  I have to migrate a bunch of torrents from one node to another hence this exercise.  

In the documentation found here https://kannibalox.github.io/pyrosimple/advanced/
There is an example shown, but when I try to run this command I get an error.

`rtxmlrpc --repr method.get '' event.download.finished`

rtxmlrpc --repr method.get '' event.download.finished
usage: rtxmlrpc [-h] [--print-completion {bash,zsh,tcsh}] [--version] [-q] [--debug] [-v] [-U URL] [-o {pretty,repr,json}] [-i] [--session] [--repl]
                [method] [args ...]
rtxmlrpc: error: unrecognized arguments: --repr

I was basically reading up on pyrocore's documentation to do some things and further ran into this issue.

https://rtorrent-docs.readthedocs.io/en/latest/cmd-ref.html#p-commands

Can you let me know if I am doing something wrong?

**Comment by kannibalox** [comment]
Aha, I didn't realize that page was actually accessible, it hasn't been fully converted for the new setups. I've fixed up that particular problem, which is that `--repr` has been replaced by `--output-format repr`. I've also removed the extremely out of data pyrotorque section, which is now here: https://kannibalox.github.io/pyrosimple/pyrotorque-tutorial/. I don't see anything else obviously wrong, but neither have I gone through and done a full test of each section.

**Comment by govardha** [comment]
Hello @kannibalox 

Thank you for the quick fix, can you please explain the usage of rtxmlrpc trying to get the p.id and p.port given the hash?  All my attempts are coming up with a null set result.  Here's my attempt.

`rtxmlrpc p.multicall $hash "" p.id= p.port=`

**Reply by kannibalox** [reply, marked as answer]
`p.multicall=` (or any of the `p.*` commands) will only return data for actively connected peers, so for most of your torrents it will probably truly be an empty set. If you want to get the full view of all your peers, you can do:
```
rtxmlrpc d.multicall2 '' main d.hash= p.multicall=,p.id=,p.port= -o repr
```

**Comment by govardha** [comment]
Thank you very much kannibalox, makes a lot of sense.  One final question, in this thread.   Please note, I LOVE all your effort and I am learning so much about python just reading your code so please don't take my questions the wrong way.  

I am looking at this link

https://pyrocore.readthedocs.io/en/latest/howto.html#adding-category-views-to-the-rtorrent-ui

When I tried out the command, I get no results, but when I try it out with a wildcard I get all the torrents that are loaded.

`rtcontrol custom_1=\! `

I am just curious what is different in how the OG rtcontrol does the filtering vs your new shiny one.

Thanks again!

**Reply by kannibalox** [reply]
This is a variant of how empty string values are no longer supported, now explicitly providing an empty string is required:
  ```bash                                                                                                                                                                             
  # Old                                                                                                                                                                               
  rtcontrol message=                                                                                                                                                                  
  rtcontrol custom_1=\!                                                                                                                                                               
  # New                                                                                                                                                                               
  rtcontrol message=\"\"                                                                                                                                                              
  rtcontrol custom_1\!=\"\"                                                                                                                                                           
  rtcontrol 'custom_1!=""' # The entire filter can also be single quoted                                                                                                              
  ```
I've added this example to the migration documentation
